_studio/shared/src/mfx_umc_mjpeg_dec.cpp

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2018 Intel Corporation
```

Comment:
I like this separation better than what we had before. Though I am not sure in the file name. I don't think this is full mjpeg decoder... that's just some fraction. Need a comment from someone else...

@dmitryermilov : can you comment?

Reply:
Due to HW limitations, in common case HW JPEG decoder cannot produce output in formats like NV12, YUY2 or RGB32. Instead, it uses some formats like YUV420, YUV422:H, YUV411 and so on. That is why MediaSDK HW JPEG decoder was extended by internal HW VPP functionality to convert YUV*** to NV12, YUY2 or RGB32. In other words, MediaSDK HW JPEG decoder consists of 2 parts: HW JPEG decoder itself, and HW VPP module. Decoder implementation is located in _studio/shared/umc/codec/jpeg_dec/src/umc_mjpeg_mfx_decode_hw.cpp. The code that you moved to separate file relates to VPP only, so my suggestion is to use name like mfx_umc_mjpeg_vpp.cpp

Reply:
@dmitrybrazhkin, Thanks for feedback!

Reply:
IMO, name like [mfx_umc_mjpeg_alloc_wrapper.cpp] would be better, since it's still UM allocator wrapper despite it does color conversion also.